### DISCUSSION TITLE
how to config eglot-workspace-configuration with multi options

### DISCUSSION THREAD

**Discussion by tshu-w** [joaotavora/eglot#875] [Q&A] [answered]
Hi, I tried something like but failed:

```emacs-lisp
(setq-default eglot-workspace-configuration
              '((:python . ((:venvPath "/data/username/.local/share/conda/envs")
                            (:analysis (:stubPath "/data/username/.local/lib/python-type-stubs"))))))
```

```
error in process filter: jsonrpc--json-encode: Wrong type argument: consp, nil
error in process filter: Wrong type argument: consp, nil
```

**Comment by tshu-w** [comment, marked as answer]
Seem like it should be: 
```emacs-lisp
  (setq-default eglot-workspace-configuration
                '((:python . (:venvPath
                              "/data/username/.local/share/conda/envs"
                              :analysis
                              (:stubPath "/data/username/.local/lib/python-type-stubs")))))
```

More example in doc will help user to better customize. I found a lot of wrong examples on github.

**Reply by joaotavora** [reply]
I recommend, instead of `setq-default`, to do
```
(add-hook 'python-mode-hook
   (lambda () (setq-local eglot-workspace-configuration ...)))
```

Everything else looks fine, as this configuration is specific to a given server.

**Reply by tshu-w** [reply]
@joaotavora Yes, local variables should be used. But I have a strange problem with tramp, local variables don't work but global variables do.

Update: I've updated the specifics of the issue and would appreciate it if you could help me look at it. https://github.com/joaotavora/eglot/discussions/876
